### PR TITLE
Spline tutorial on exporting a spline design and animating in remotion

### PR DESCRIPTION
**Pull request by pabloescoder**
References #1376 
Added a comprehensive step-by-step guide for building a simple model in spline.design, modifying that model, exporting it as react-three-fiber code and using that in Remotion to create an animated video. 
Attached images wherever possible for better understanding, along with a video demonstrating the final animation that was built in the guide, and added code snippets wherever required.

Markdown file with images and proper formatting can be viewed here: https://github.com/remotion-dev/remotion/blob/e794e0ef43e00e166cc754642397a53964455e94/packages/docs/docs/spline.md

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#1376: Write a tutorial on how to import a design from spline.design into Remotion](https://issuehunt.io/repos/274495425/issues/1376)
---
</details>

### CODE REVIEW

File: packages/docs/docs/spline.md [hunk 1]
```diff
@@ -201,15 +201,7 @@ RTF = React Three Fiber
...
    ```tsx
-   import React, { useEffect } from "react";
-   import { useThree } from "@react-three/fiber";
-   import {
-     interpolate,
-     spring,
-     useCurrentFrame,
-     useVideoConfig,
-   } from "remotion";
-   import { CAMERA_DISTANCE } from "./helpers/layout";
```
Comment (pabloescoder) [inline packages/docs/docs/spline.md]: Were these import statements and the one at line 290 causing the Vercel build to fail?

Comment (JonnyBurger) [reply packages/docs/docs/spline.md]: I think it was `style="color: red"`. Fixed that for you!

**Comment by JonnyBurger** [comment]
Looks totally awesome! Impressed by how thorough this is. I need to take some time to review this later, stay tuned!
I increased the bounty to $175 since you definitely put some effort into this :)

**Comment by pabloescoder** [comment]
That's wonderful! Glad to hear that :D 
I had a lot of fun while writing this tutorial, and I love how simple it is to create videos using Remotion. I'm definitely going to check out Spline as well as Remotion in depth.

File: packages/docs/docs/spline.md (hunk 1 above)
Comment (pabloescoder) [reply packages/docs/docs/spline.md]: Perfect, thank you!

**Comment by pabloescoder** [comment]
On checking the most recent version of spline.md, I noticed that 2 code blocks are empty (maybe the code inside those was deleted by mistake?)

File: packages/docs/docs/spline.md [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+   3. Changed the background colour from #4b4d52 to #ffb703.
+   4. Reordered props. (Shorthand props must be listed before all other props) eslint [react/jsx-sort-props](https://github.com/yannickcr/eslint-plugin-react/tree/master/docs/rules/jsx-sort-props.md)
+
+8. Inside the **src** folder, create a new component called **Torus.tsx** and type/paste the above code.
+
+9. Add the following imports at the **top** of **Torus.tsx**, below the useSpline import.
+
+   ```tsx
+
+   ```
```
Comment (pabloescoder) [inline packages/docs/docs/spline.md]: Missing code:-

```tsx
import React, { useEffect } from "react";
import { useThree } from "@react-three/fiber";
import {
  interpolate,
  spring,
  useCurrentFrame,
  useVideoConfig,
} from "remotion";
import { CAMERA_DISTANCE } from "./helpers/layout";
```

File: packages/docs/docs/spline.md [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+
+    This component is our entire video. A video is made up of one or more compositions (scenes). The props of a Composition component are quite self explanatory. Change the durationInFrames prop from 300 to 210. The length of our video is calculated by durationInFrames/fps.
+
+15. Navigate to **Scene.tsx**
+
+    Remove the `import {Phone} from './Phone';` statement from the top and replace it with
+
+    ```tsx
+
+    ```
```
Comment (pabloescoder) [inline packages/docs/docs/spline.md]: Missing code: 
```tsx
import { Torus } from "./Torus";
```

**Comment by pabloescoder** [comment]
Should I add this back to spline.md?

**Comment by JonnyBurger** [comment]
@pabloescoder Sorry if I deleted those accidentally! 😫 did not mean to! Would be nice if you can add them back

**Comment by pabloescoder** [comment]
@JonnyBurger It's completely alright! I've added the missing code 😄

**Comment by JonnyBurger** [comment]
Thanks @pabloescoder, I just followed the tutorial and it's awesome!

I rewarded you already on IssueHunt!

But before we're going to merge it, are you okay with the changes I made? I deleted a lot with the goal to only keep the instructions that one has to follow but to still get the desired result. I also solved the R3F v7 issues alongside the way!

I will let someone else follow the tutorial before we merge it to make sure it is clearly understandeable. @MehmetAdemi do you want to try to build a 3D scene in Remotion using this tutorial?

@pabloescoder For statistics, how long do you think you spent on this issue?

**Comment by pabloescoder** [comment]
Thank you so much @JonnyBurger!

I am happy with all your changes, and I completely understand why you removed the redundant parts. The tutorial looks much better now. I'm glad you sorted out the R3F v7 issue!

Could you also add your name to the author field at the top of the markdown file? I don't want to take credit for everything considering your time spent improving the readme and solving any issues. 

The final video is not showing up when I view the markdown file on GitHub and locally. Is there a link where I can check out the tutorial on the Remotion website? 

@JonnyBurger It took me around 8 hours (not counting breaks) to write the tutorial and generate the final video. Mainly as this was my first time using Spline and Remotion, I spent quite some time learning the basics of both before getting started with the tutorial. 
Please let me know if changes are needed or if any additional work is required from my side!

**Comment by JonnyBurger** [comment]
@pabloescoder As you wish, added myself!
Thank you for your huge effort, added the time spent to our spreadsheet.

Here you can preview the branch: https://remotion-git-fork-pabloescoder-spline-tutorial-remotion.vercel.app/docs/spline

**Comment by pabloescoder** [comment]
@JonnyBurger Wonderful! The tutorial looks amazing 😄

**Comment by JonnyBurger** [comment]
We did not yet have time to follow the tutorial ourselves, but it's on our TODO list!
I don't want to hold it back, so let's release it 🥳

**Review by JonnyBurger — APPROVED** [review]

**Comment by pabloescoder** [comment]
Sounds great! :D

**Pull request merged**